### PR TITLE
[ADD] project_stage_warning: Show warning when project task stage skipped

### PR DESCRIPTION
**Pull request by wardm95**

### CODE REVIEW

File: project_stage_warning/models/project_task.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    def write(self, values):
+        if values.get('stage_id'):
+            new_stage = self.env['project.task.type'].browse(values.get('stage_id'))
+            end_sequence = new_stage.sequence
+            for task in self:
+                start_sequence = task.stage_id.sequence
+                skipped_stage = self.env['project.task.type'].search([
```
Comment (Yenthe666) [inline project_stage_warning/models/project_task.py]: `search_count` for perf?

File: project_stage_warning/models/project_task.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    def write(self, values):
+        if values.get('stage_id'):
+            new_stage = self.env['project.task.type'].browse(values.get('stage_id'))
```
Comment (Yenthe666) [inline project_stage_warning/models/project_task.py]: Check if you don't need `sudo()`

**Review by Yenthe666 — CHANGES_REQUESTED** [review]
feedbacks

File: project_stage_warning/models/project_task.py (hunk 1 above)
Comment (wardm95) [reply project_stage_warning/models/project_task.py]: With `search_count` we don't have the name of the stage to show in the warning.
We might use `search_read` instead?

File: project_stage_warning/models/project_task.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    def write(self, values):
+        if values.get('stage_id'):
+            new_stage = self.env['project.task.type'].sudo().browse(values.get('stage_id'))
+            end_sequence = new_stage.sequence
+            for task in self:
+                start_sequence = task.stage_id.sequence
+                skipped_stage = self.env['project.task.type'].sudo().search([
+                    ('sequence', '<', end_sequence),
+                    ('sequence', '>', start_sequence),
+                    ('project_ids', '=', task.project_id.id),
```
Comment (sswapnesh) [inline project_stage_warning/models/project_task.py:18]: task.project_id.id

we might need to check if project is updated on task or not

```suggestion
                    ('project_ids', '=', values.get('project_id', task.project_id.id)),

```

Comment (wardm95) [reply project_stage_warning/models/project_task.py:18]: Good catch, i've updated it as you suggested

**Pull request merged**